pkg/geo/geomfn/distance.go

```diff
@@ -913,61 +939,87 @@ pointOuterLoop:
 			// Each point in a (multi)point must intersect a polygon in the
 			// (multi)point to be covered by it.
 			return false, nil
+		case PointPolygonWithin:
```

Comment:
yeah i uh look forward to the refactor, this was difficult to reason about :( 
too many fallthroughs and continues...